automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/azure/AzureAutopauseSpec.scala

```diff
@@ -0,0 +1,105 @@
+    with TableDrivenPropertyChecks
+    with Retries
+    with CleanUp {
+
+  implicit val accessToken: IO[AuthToken] = Hermione.authToken()
+
+  "azure runtime autopauses" taggedAs ExcludeFromJenkins in { workspaceDetails =>
+    val workspaceId = workspaceDetails.workspace.workspaceId
+
+    val labelMap: java.util.HashMap[String, String] = new java.util.HashMap[String, String]()
+    labelMap.put("automation", "true")
+
+    val runtimeName = randomClusterName
+    val res =
+      for {
+        _ <- loggerIO.info(s"AzureAutoPauseSpec: About to create runtime")
+        runtimeClient <- GeneratedLeonardoClient.generateRuntimesApi
+
+        // autopause set to 15 minutes to make sure metrics monitor call is ignored (doesn't reset the clock)
+        createReq = new CreateAzureRuntimeRequest()
+          .labels(labelMap)
+          .machineSize("Standard_DS1_v2")
+          .disk(
+            new AzureDiskConfig()
+              .name(generateAzureDiskName())
+              .size(50)
+              .labels(labelMap)
+          )
+          .autopauseThreshold(15)
+
+        _ <- IO(runtimeClient.createAzureRuntime(workspaceId, runtimeName.asString, false, createReq))
+        _ <- loggerIO.info(s"AzureAutoPauseSpec: Create runtime request submitted. Starting to poll GET")
+
+        // ------------------ Waiting for Create ------------------ //
+
+        // Verify the initial getRuntime call
+        callGetRuntime = IO(runtimeClient.getAzureRuntime(workspaceId, runtimeName.asString))
+
+        initialGetRuntimeResponse <- callGetRuntime
+        _ <- loggerIO.info(s"initial get runtime response $initialGetRuntimeResponse")
+        _ = initialGetRuntimeResponse.getStatus shouldBe ClusterStatus.CREATING
+
+        _ <- loggerIO.info(
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} in creating status detected"
+        )
+
+        // Verify the runtime eventually becomes Running (in 20 minutes)
+        monitorCreateResult <- streamUntilDoneOrTimeout(
+          callGetRuntime,
+          120,
+          10 seconds,
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} did not finish creating after 20 minutes"
+        )(implicitly, GeneratedLeonardoClient.runtimeInStateOrError(ClusterStatus.RUNNING))
+
+        _ <- loggerIO.info(
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} create monitor result: $monitorCreateResult"
+        )
+        _ = monitorCreateResult.getStatus() shouldBe ClusterStatus.RUNNING
+
+        // ------------------ Waiting for Autopause ------------------ //
+
+        // new IO for new stream
+        callGetRuntimeStopping = IO(runtimeClient.getAzureRuntime(workspaceId, runtimeName.asString))
+
+        _ <- loggerIO.info(
+          s"AzureAutoPauseSpec: runtime $workspaceId/${runtimeName.asString} waiting to autopause"
+        )
+
+        _ <- IO.sleep(20 minutes) // sleep for 20 minutes before checking if the runtime paused
```

Comment:
I found that it wasn't autopausing when I polled for some reason...but I also changed a couple other things between so I can try adding that back and testing again during my meetings today

Reply:
Yeah, I would add polling, because the listener is supposed to ignore that